cloudconfig/cloudinit/cloudinit.go

```diff
@@ -325,6 +325,20 @@ func (cfg *cloudConfig) SetSSHAuthorizedKeys(rawKeys string) {
 	}
 }
 
+// SetSSHKeys is defined on the SSHKeysConfig interface.
+func (cfg *cloudConfig) SetSSHKeys(keys SSHKeys) {
+	attrs := make(map[string]interface{})
+	if keys.RSA != nil {
+		attrs[string(RSAPrivate)] = keys.RSA.Private
+		attrs[string(RSAPublic)] = keys.RSA.Public
+	}
+	if len(attrs) > 0 {
+		cfg.SetAttr("ssh_keys", attrs)
+	} else {
+		cfg.UnsetAttr("ssh_keys")
+	}
+}
```

Comment:
The idea was the make room for other key types, but you're right. I'll just keep it simple and we can expand later if needed.